include/dxc/DXIL/DxilModule.h

```diff
@@ -345,7 +349,12 @@ class DxilModule {
   unsigned m_ActiveStreamMask = 0;
 
   enum IntermediateFlags : uint32_t {
+
     LegacyResourceReservation = 1 << 0,
+    UnusedResourceBindingMask =
+        7 << 1, // 3 bit reserved for unused resource binding
```

Comment:
I think it would be easier to fold `LegacyResourceReservation` into `UnusedResourceBinding::ReserveExplicit`? That way you can simply serialize that state and get rid of `IntermediateFlags`.